muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/AgentCachingPoolStrategy.java

```diff
@@ -461,6 +505,45 @@ public TypeList.Generic getInterfaces() {
         }
         return cachedInterfaces;
       }
+
+      private class LazyAnnotationMethodDescription extends DelegatingMethodDescription {
+
+        LazyAnnotationMethodDescription(MethodDescription.InDefinedShape method) {
+          super(method);
+        }
+
+        @Override
+        public AnnotationList getDeclaredAnnotations() {
+          // Run getDeclaredAnnotations with ThreadLocal. ThreadLocal helps us detect types looked
+          // up by getDeclaredAnnotations and treat them specially.
+          enterLoadAnnotations();
+          try {
+            return method.getDeclaredAnnotations();
+          } finally {
+            exitLoadAnnotations();
+          }
```

Comment:
@trask you asked how big of a difference this makes. Calling `method.getDeclaredAnnotations()` without  `enterLoadAnnotations` produced `count=541790, time=1290ms` and with `enterLoadAnnotations` `count=541715, time=87ms` when starting liferay.